src/scene/scene.cpp

```diff
@@ -215,9 +217,64 @@ void Scene::sortTopologically() {
 }
 
 void Scene::initialize() {
```

Comment:
FYI, In my feature/data-management branch, `Scene::initialize` does not exist in this form. Instead, nodes are initialized whenever they are added to the scene. I'm sure it's possible for these changes to be incorporated in there anyway though.